inference-engine/include/ie_extension.h

```diff
@@ -105,24 +100,32 @@ class INFERENCE_ENGINE_API_CLASS(Extension) : public IExtension {
     details::SOPointer<IExtension> actual;
 };
 
+// IE_SUPPRESS_DEPRECATED_START
 /**
- * @brief Creates a special shared_pointer wrapper for the given type from a specific shared module
- *
- * @param name A std::string name of the shared library file
- * @return shared_pointer A wrapper for the given type from a specific shared module
+ * @brief Creates extension using deprecated API
+ * @tparam T extension type
+ * @param name extension library name
+ * @return shared pointer to extension
  */
-template <>
-inline std::shared_ptr<IExtension> make_so_pointer(const std::string& name) {
+template<typename T = IExtension>
+// INFERENCE_ENGINE_DEPRECATED("Use std::make_shared<Extension>")
```

Comment:
Why did you add deprecation macros as comments?

Reply:
Just to pass CI as we has some builds that use `Werror` flag. Should be uncommented before merge.

Reply:
@apankratovantonp release branch has been created, so we are ready to merge this change. Can you enable deprecation warning back?